packages/devextreme/js/__internal/viz/chart_components/rolling_stock.ts

```diff
@@ -0,0 +1,106 @@
+interface BoundingRect {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+interface RollingStockLabel {
+  getBoundingRect: () => BoundingRect;
+  hideInsideLabel: (coords: { x: number; y: number }) => boolean;
+  getData: () => { value: number | string | Date };
```

Comment:
i mb missed something. Whats the point of object having 1 field? i'm talking about 'getData' return value . can't we return value right away? And then we can refactor a little to make code more readable and reusable

```suggestion
type StockLabelData = number | string | Date;
interface Coordinates {
  x: number;
  y: number;
}

interface Bounds {
  width: number;
  height: number;
}

type BoundingRect = Coordinates & Bounds;

interface RollingStockLabel {
  getBoundingRect: () => BoundingRect;
  hideInsideLabel: (coords: Coordinates) => boolean;
  getData: () => StockLabelData;
```
also i suppose it's possible to use this new 'Coordinates' type for 'shift' function if we change param type

Reply:
in RollingStock we use [label of series](https://github.com/DevExpress/DevExtreme/blob/23_2/packages/devextreme/js/__internal/viz/chart_components/m_base_chart.ts#L133). And we do not change API of label